great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -2411,6 +2411,13 @@ def _sqlalchemy_map_condition_query(
     domain_kwargs: dict = dict(**compute_domain_kwargs, **accessor_domain_kwargs)
     result_format: dict = metric_value_kwargs["result_format"]
 
+    # We will not return map_condition_query if return_unexpected_index_query = False
+    return_unexpected_index_query: Union[bool, None] = result_format.get(
```

Comment:
general question: Why do we need this Union type here? Woudn't python's truthiness take into account whether something is `False` or `None` and wouldn't it have then same effect?

Reply:
I think you're totally right :) Thanks for catching this @rdodev